aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,273 @@
+        inputs : dict
+            The inputs dictionary.
+
+        port_namespace : `aiida.engine.processes.ports.PortNamespace`
+            An instance of aiida's `PortNameSpace`.
+
+        Returns
+        -------
+        str or None
+            Error message if validation fails, None otherwise.
+        """
+        # Wrapping processes may choose to exclude certain input ports
+        # If the ports have been excluded, skip the validation.
+        if any(key not in port_namespace for key in ("calctype", "structure")):
+            return None
+
+        for key in ("calctype", "structure"):
+            if key not in inputs:
+                return f"required value was not provided for the `{key}` namespace."
+
+        valid_calctypes = {"singlepoint", "geom opt"}
+        if "calctype" in inputs:
+            if str(inputs["calctype"].value) not in valid_calctypes:
+                return f"The 'calctype' must be one of {valid_calctypes}, \
+                    but got '{inputs['calctype']}'."
+
+        if "input_filename" in inputs:
+            if not str(inputs["input_filename"].value).endswith(".cif"):
+                return "The parameter 'input_filename' must end with '.cif'"
+
+        # If both structure and calctype are provided, return None
+        return None
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files for the `Calcjob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        # Create needed inputs
+        # Define architecture from model if model is given,
+        # otherwise get architecture from inputs and download default model
+        architecture = (
+            str((self.inputs.model).architecture)
+            if self.inputs.model
+            else str(self.inputs.architecture.value)
+        )
+        model_path = (
+            str((self.inputs.model).filepath)
+            if self.inputs.model
+            else ModelData.download(
+                "https://github.com/stfc/janus-core/raw/main/tests/models/mace_mp_small.model",  # pylint:disable=line-too-long
+                architecture,
+            ).filepath
+        )
+
+        # The inputs are saved in the node, but we want their value as a string
+        calctype = str((self.inputs.calctype).value)
+        precision = str((self.inputs.precision).value)
+        device = str((self.inputs.device).value)
+        xyz_filename = str((self.inputs.xyz_output_name).value)
+        input_filename = self.inputs.metadata.options.input_filename
+        log_filename = str((self.inputs.log_filename).value)
```

Comment:
Seems awkward to be constantly casting to strings. Would recommend adding getter `@property`s on the object for these. 

```python
@property
def calctype(self):
    return str(self.inputs.calctype.value)
```

Reply:
good point and I don't even need the str actually (value is already a str) , so I'll remove it